isisd/isis_tlvs.c

```diff
@@ -3593,6 +4527,23 @@ void isis_tlvs_set_dynamic_hostname(struct isis_tlvs *tlvs,
 		tlvs->hostname = XSTRDUP(MTYPE_ISIS_TLV, hostname);
 }
 
+/* Set Router Capability TLV parameters */
+void isis_tlvs_set_router_capability(struct isis_tlvs *tlvs,
+				     struct in_addr router_id,
+				     const struct isis_router_cap *cap)
+{
+	XFREE(MTYPE_ISIS_TLV, tlvs->router_cap);
+	tlvs->router_cap = XCALLOC(MTYPE_ISIS_TLV, sizeof(*tlvs->router_cap));
+	tlvs->router_cap->router_id = router_id;
+	tlvs->router_cap->srgb.flags =
+		ISIS_SUBTLV_SRGB_FLAG_I | ISIS_SUBTLV_SRGB_FLAG_V;
+	tlvs->router_cap->srgb.range_size = cap->srgb.range_size;
+	tlvs->router_cap->srgb.lower_bound = cap->srgb.lower_bound;
+	for (int i = 0; i < SR_ALGORITHM_COUNT; i++)
+		tlvs->router_cap->algo[i] = cap->algo[i];
+	tlvs->router_cap->msd = cap->msd;
```

Comment:
nit: you could copy the whole structure in a single line like this: `*tlvs->router_cap = *cap;`.

Reply:
done